src/Illuminate/Contracts/Validation/DataAwareRule.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Illuminate\Contracts\Validation;
+
+interface DataAwareRule
+{
+    /**
+     * Set the data under validation.
+     *
+     * @param  array  $data
+     * @return $this
+     */
+    public function setData($data);
```

Comment:
maybe `ArrayAccess` would also be valid here.

Reply:
The given `$data` will be always an array type:  https://github.com/laravel/framework/blob/99172a821667ee462644721dc7d6c8a07a88293f/src/Illuminate/Validation/Validator.php#L83.